snapcraft/_store.py

```diff
@@ -281,3 +282,82 @@ def download(snap_name, channel, download_path, arch):
         raise RuntimeError(
             'Failed to download {} at {} (mismatched SHA)'.format(
                 snap_name, download_path))
+
+
+def _get_snap_data(name, store):
+    search_results = store.cpi.search_package(name, 'stable', None)
+    if search_results is None:
+        raise RuntimeError(
+            'Snap {name} cannot be found'.format(
+                name=name))
+    details_url = search_results['_links']['self']['href']
+    details = store.cpi.get(details_url).json()
+    if details.get('result') == 'error':
+        raise RuntimeError(
+            'Error obtaining details for snap {}: {}'.format(
+                name, ', '.join(details['errors'])))
+    return details
+
+
+def gated(snap_name):
+    """Print list of snaps gated by snap_name."""
+    store = storeapi.StoreClient()
+    # Get data for the gating snap
+    snap_data = _get_snap_data(snap_name, store)
+    snap_id = snap_data['snap_id']
+    validations = store.get_validations(snap_id)
+
+    table_data = []
+    for v in validations:
+        table_data.append([v['approved-snap-name'], v['approved-snap-revision']])
+    tabulated = tabulate(table_data, headers=['Name', 'Approved'],
+                         tablefmt="plain")
+    print(tabulated)
+
+
+def validate(snap_name, validations, revoke=False, key=None):
+    """Generate, sign and upload validation assertions."""
+    store = storeapi.StoreClient()
+
+    # Get data for the gating snap
+    snap_data = _get_snap_data(snap_name, store)
+    release = str(snap_data['release'][0])
+    snap_id = snap_data['snap_id']
+    developer_id = snap_data['developer_id']
+    # Then, for each requested validation, generate assertion
+    # XXX FIXME: ensure validations are of the form name=number
+    for validation in validations:
+        gated_name, rev = validation.split('=', 1)
+        approved_data = _get_snap_data(gated_name, store)
+        assertion = {
+            'type': 'validation',
+            'authority-id': developer_id,
+            'series': release,
+            'snap-id': snap_id,
+            'approved-snap-id': approved_data['snap_id'],
+            'approved-snap-revision': rev,
+            'timestamp': datetime.datetime.utcnow().isoformat() + 'Z',
+            'valid': "true"
+        }
+        if revoke:
+            assertion['valid'] = "false"
+
+        cmdline = ['snap', 'sign']
+        if key is not None:
```

Comment:
There is a remote chance of someone doing something like --key='' I guess but that ends up being None. Changing it, since empty strings are invalid key names anyway.